pkg/reconciler/instances/istio/istio.go

```diff
@@ -24,13 +41,93 @@ func init() {
 		log.Fatalf("Could not create '%s' component reconciler: %s", ReconcilerName, err)
 	}
 
-	commander := istioctl.DefaultCommander{}
+	pathsConfig := os.Getenv(istioctlBinaryPathEnvKey)
+	paths, err := parsePaths(pathsConfig)
+	if err != nil {
+		log.Fatalf("Could not create '%s' component reconciler: Error parsing env variable '%s': %s", ReconcilerName, istioctlBinaryPathEnvKey, err.Error())
+	}
+
+	//istioctlResolver, err := istioctl.NewDefaultIstioctlResolver(paths, istioctl.DefaultVersionChecker{})
+	commanderResolver, err := newDefaultCommanderResolver(paths, log)
+	if err != nil {
+		log.Fatalf("Could not create '%s' component reconciler: %s", ReconcilerName, err.Error())
+	}
+
 	gatherer := data.NewDefaultGatherer()
 	matcher := pod.NewParentKindMatcher()
 	provider := clientset.DefaultProvider{}
 	action := reset.NewDefaultPodsResetAction(matcher)
 	istioProxyReset := proxy.NewDefaultIstioProxyReset(gatherer, action)
-	performer := actions.NewDefaultIstioPerformer(&commander, istioProxyReset, &provider)
+	performer := actions.NewDefaultIstioPerformer(commanderResolver, istioProxyReset, &provider)
 	reconciler.WithReconcileAction(NewReconcileAction(performer)).
 		WithDeleteAction(NewUninstallAction(performer))
 }
+
+//Provides runtime wiring between istioctl.DefaultVersionChecker, istioctl.DefaultIstioctlResolver and istioctl.DefaultCommander
+//Implements actions.CommanderResolver
+type defaultCommanderResolver struct {
+	log                 *zap.SugaredLogger
+	paths               []string
+	istioBinaryResolver istioctl.IstioctlResolver
+}
+
+func (dcr *defaultCommanderResolver) GetCommander(version istioctl.Version) (istioctl.Commander, error) {
+	istioBinary, err := dcr.istioBinaryResolver.FindIstioctl(version)
+	if err != nil {
+		return nil, err
+	}
+
+	dcr.log.Debugf("Resolved istioctl binary: Requested: %s, Found: %s", version.String(), istioBinary.Version().String())
+
+	res := istioctl.NewDefaultCommander(*istioBinary)
+	return &res, nil
+}
+
+func newDefaultCommanderResolver(paths []string, log *zap.SugaredLogger) (actions.CommanderResolver, error) {
+
+	istioBinaryResolver, err := istioctl.NewDefaultIstioctlResolver(paths, istioctl.DefaultVersionChecker{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &defaultCommanderResolver{
+		log:                 log,
+		paths:               paths,
+		istioBinaryResolver: istioBinaryResolver,
+	}, nil
+}
+
+//The input must contain a list of full/absolute filesystem paths of istioctl binaries.
+//Entries must be separated by a colon character ':'
+func parsePaths(input string) ([]string, error) {
```

Comment:
Could we check that every result path exists and it makes sense to forward it on?